crates/vector-store/src/index/factory.rs

```diff
@@ -22,4 +29,6 @@ pub trait IndexFactory {
         expansion_search: ExpansionSearch,
         space_type: SpaceType,
     ) -> anyhow::Result<mpsc::Sender<Index>>;
+    fn get_type(&self) -> Type;
+    fn get_version(&self) -> String;
```

Comment:
```suggestion
    fn version(&self) -> String;
```
IMHO we don't need type of index. In Rust the idiomatic getter is without `get_` prefix. Idiomatic setter is with `set_` prefix.

Reply:
Naming fixed. Engine type removed. Now API returns only engine version.